src/simoc_sam/csv_writer.py

```diff
@@ -0,0 +1,84 @@
+import os
+import csv
+import json
+
+from simoc_sam.sensors import utils
+
+import paho.mqtt.client as mqtt
+
+
+HOST = 'samrpi1.local'
+PORT = 1883
+KEEPALIVE = 10  # in seconds
+TOPIC = "sam/#" 
+
+# Callback when the client connects to the broker
+def on_connect(client, userdata, flags, rc, properties=None):
+    print("Connected with result code " + str(rc))
+    # Subscribe to the MQTT topic
+    client.subscribe(args.topic)
+
+def on_message(client, userdata, msg):
+    payload = msg.payload.decode("utf-8")
+    topic = str(msg.topic)
+    print(f"Received message: {payload}")
+    print(f"from topic: {topic}")
+    # Parse the payload (assuming it's JSON, adjust as needed)
+    data = json.loads(payload)
+
+    # Define file name based on topic
+    csv_file_path = "/home/sam/data/" + topic.replace("/", "_").replace("#", "") + ".csv"
+
+    # Append the data to the CSV file
+    with open(csv_file_path, mode='a', newline='') as csv_file:
+        csv_writer = csv.writer(csv_file)
+        #field_names = sorted(data.keys())
+        field_names = ['n', 'timestamp', *[k for k in data if k not in ('n', 'timestamp')]]
+        # Check if the CSV file is empty
+        is_empty = os.stat(csv_file_path).st_size == 0
+
+        # Write headers if the file is empty
+        if is_empty:
+            # Update the set of field names based on the payload keys
+            csv_writer.writerow(field_names)
+
+        # Write data to the CSV file
+        csv_writer.writerow([data.get(field, '') for field in field_names])
+
+    #print("Data logged to CSV")
+
+# main
+
+def main(host=HOST, port=PORT, topic=TOPIC):
```

Comment:
I don't think this works.  These values are always overridden by the `args.*` ones.